Added CovidInfo Project

jaivsh opened the pull request:
## Description
I've added the covid data scrapper to the reop along withproper documentation. 
For the purpose of checking/running/executing the scrapper, I've also created a colab file here: [Colab notebook link](https://colab.research.google.com/drive/1uD11Vp1AjoBsdZJCykQGNNCEoSRknRzt?usp=sharing)

> The changes made in this pull request should be briefly described.

## Resolves: #463 

## Checklist

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots
The relevant screenshots for each function are as follows: 


1. Scrape():
<img width="1470" alt="Screenshot 2023-07-12 at 9 12 58 PM" src="https://github.com/Clueless-Community/scrape-up/assets/93859359/9a36c728-4ec6-49d4-8ad2-f60810ac3bbb">

2. findCountry():
<img width="1470" alt="Screenshot 2023-07-12 at 9 14 00 PM" src="https://github.com/Clueless-Community/scrape-up/assets/93859359/fee5abbf-b03c-4e89-b0bd-fa8c060f96b1">

3. sortbycases():
<img width="1470" alt="Screenshot 2023-07-12 at 9 20 01 PM" src="https://github.com/Clueless-Community/scrape-up/assets/93859359/c90fbfee-f67f-4358-b5f2-d87dc3e9ac9c">

4. sortbydeaths():
<img width="1470" alt="Screenshot 2023-07-12 at 9 21 32 PM" src="https://github.com/Clueless-Community/scrape-up/assets/93859359/b9756822-a30a-4cc2-b399-f3bb48495eea">

5. totalcases():
<img width="1470" alt="Screenshot 2023-07-12 at 9 22 29 PM" src="https://github.com/Clueless-Community/scrape-up/assets/93859359/4bb9a7a4-a9df-43b6-b9a1-142bd069bcab">

6. totaldeaths():
<img width="1470" alt="Screenshot 2023-07-12 at 9 25 39 PM" src="https://github.com/Clueless-Community/scrape-up/assets/93859359/2a6bbb6d-92b2-478c-8b96-b83c314bd02e">

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

jaivsh commented:
@nikhil25803 I've completed the implementation. PLease check out this PR

nikhil25803 commented on src/scrape_up/CovidInfo/covid_info.py:
```diff
@@ -0,0 +1,88 @@
+        ])
+
+      except StopIteration:
+        break
```
On exception, return `None`. It helps the user to script some conditions on it.

nikhil25803 commented on src/scrape_up/CovidInfo/covid_info.py:
```diff
@@ -0,0 +1,88 @@
+    data = self.scrape()
+    data.remove(['Country','Number of Cases', 'Deaths'])
+    data = sorted(data, key= lambda r:r[1],reverse=rev)
+    return data
```
Implement try-exception in all the methods.

jaivsh replied:
Sir, I've implemented the changes as you prescribed. Please review the work. Thanks!

nikhil25803 commented on src/scrape_up/CovidInfo/covid_info.py:
```diff
@@ -0,0 +1,88 @@
+    soup = BeautifulSoup(page.text, 'html.parser')
+    req = soup.find_all('a',{"href": "/coronavirus/coronavirus-death-toll/"})
+    k = req[0].find('strong').contents[0].split(' ')
+    return k[0]
```
Write a docstring for the class and also for each method

nikhil25803 reviewed (changes requested):
Requested some changes @jaivsh. Implement them and update the PR.

jaivsh commented:
@nikhil25803 I've performed the following changes:
- added docstring for class and functions
- added try/catch for all functions
- added return None in required places. 

Note: In first function scrape(), i initially inserted the ```except StopIteration: break``` so that when all the rows would be appended to list, and there would be no more rows, the code should run into StopIteration exception and would execute the catch block, where it would encounter break and would stop the loop. So, it is difficult to add a try/except block in first function. However, I've added it's equivalent which would ```return Null``` in case no row is added to the list.
